web3.js/test/program-tests/vote.test.ts

```diff
@@ -167,6 +167,24 @@ describe('VoteProgram', () => {
 
       // Withdraw from Vote account
       let recipient = Keypair.generate();
+      const voteBalance = await connection.getBalance(newVoteAccount.publicKey);
+      const voteBalanceAfterWithdraw = voteBalance - LAMPORTS_PER_SOL;
+
+      expect(
+        VoteProgram.safeWithdraw(
+          {
+            votePubkey: newVoteAccount.publicKey,
+            authorizedWithdrawerPubkey: authorized.publicKey,
+            lamports: LAMPORTS_PER_SOL,
+            toPubkey: recipient.publicKey,
+          },
+          voteBalanceAfterWithdraw,
```

Comment:
How about using `voteBalance` here to trigger the exact error condition you want to test?
```suggestion
          {
            votePubkey: newVoteAccount.publicKey,
            authorizedWithdrawerPubkey: authorized.publicKey,
            lamports: voteBalance,
            toPubkey: recipient.publicKey,
          },
          voteBalance,
```

Reply:
I think we want to keep the second parameter as `voteBalanceAfterWithdraw` since it removes the minimum amount needed for rent which is how we'll be using this.  And since `voteBalance` will be greater than that it will trigger the exception.

Reply:
The whole point of the check is to make sure that you're not withdrawing too much, correct? And you'll only withdraw too much if you remove everything in the account.  If the account is left with some amount `0 < lamports < rent_exemption`, the instruction will fail, since new rent-paying accounts aren't allowed.

Reply:
No, you can still withdraw too much without draining the account.  If you withdraw an amount that is greater than the `current balance of the account - minimum rent` it will still become unusable.  That's what happened to me and it was awful.  So the whole point is to make sure you leave around enough for rent after withdrawing.

Reply:
That being said, your suggestion would indeed trigger the exception we're testing for so it was valid.  But I'm trying to show users what (in my experience) I'm looking for going forward.

Reply:
Yes it was

Reply:
In that case, how about also providing the rent exempt minimum for the account as a parameter? That way you can do the full check in your `safeWithdraw` function

Reply:
All set. Let me know if naming conventions are fine, thanks.